encoders/nlp/TransformerTorchEncoder/__init__.py

```diff
@@ -188,10 +224,20 @@ def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
             )
             input_tokens = {k: v.to(self.device) for k, v in input_tokens.items()}
 
-            with self.amp_accelerate():
-                outputs = getattr(self.model, self.embedding_fn_name)(**input_tokens)
+            if self.api_token is not None:
+                outputs = self._api_call(list(data))
+
+                hidden_states = torch.tensor([outputs])
+            else:
+                with self.amp_accelerate():
+                    outputs = getattr(self.model, self.embedding_fn_name)(
+                        **input_tokens
+                    )
+                    if isinstance(outputs, torch.Tensor):
+                        return outputs.cpu().numpy()
+                    hidden_states = outputs.hidden_states
 
             if not isinstance(outputs, torch.Tensor):
-                return self._compute_embedding_from_model_output(outputs, input_tokens)
+                return self._compute_embedding(hidden_states, input_tokens)
             else:
```

Comment:
I think this `if` else is not anymore needed.

Reply:
Good catch. Thanks!